### PR TITLE
refactor: types ARButton and VRButton

### PR DESCRIPTION
**Pull request by DylanTackoor**
### What

- `ARButton.js` -> `ARButton.ts`
- `VRButton.js` -> `VRButton.ts`

### Why

Types XRButtons. Didn't see an issue tracking these in any of the typing milestones so no worries if it's too early to merge this. Just wanted to fix _something_ inbetween working on typing the loaders as they're taking a while.

### Checklist

- [x] Ready to be merged

### CODE REVIEW

File: src/webxr/ARButton.ts [hunk 1]
```diff
@@ -1,28 +1,37 @@
+import { WebGLRenderer, XRSession, XRSessionInit } from 'three'
+
+interface ARButtonSessionInit extends XRSessionInit {
```
Comment (joshuaellis) [inline src/webxr/ARButton.ts:3]: Lets export this type, even if no one uses it, we might aswell.

File: src/webxr/ARButton.ts [hunk 2]
```diff
@@ -79,7 +88,8 @@ class ARButton {
 
       button.onclick = function () {
```
Comment (joshuaellis) [inline src/webxr/ARButton.ts:89]: Do you think we could use `addEventListener` instead?

File: src/webxr/ARButton.ts [hunk 3]
```diff
@@ -79,7 +88,8 @@ class ARButton {
...
         if (currentSession === null) {
-          navigator.xr.requestSession('immersive-ar', sessionInit).then(onSessionStarted)
+          const xr: THREE.XR = (navigator as any).xr
```
Comment (joshuaellis) [inline src/webxr/ARButton.ts:91]: unconventional, but i think i'd prefer a `@ts-expect-error` here instead, small description afterwards explaining that we need to add webxr types which *might* be an open issue?

File: src/webxr/VRButton.ts [hunk 4]
```diff
@@ -38,16 +30,13 @@ class VRButton {
...
       button.textContent = 'ENTER VR'
 
-      button.onmouseenter = function () {
-        button.style.opacity = '1.0'
-      }
-
-      button.onmouseleave = function () {
-        button.style.opacity = '0.5'
-      }
+      button.onmouseenter = () => (button.style.opacity = '1.0')
```
Comment (joshuaellis) [inline src/webxr/VRButton.ts:33]: same as above with event listeners, what do you think?

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Few discussion points, nothing major imo

File: src/webxr/ARButton.ts (hunk 2 above)
Comment (DylanTackoor) [reply src/webxr/ARButton.ts:89]: yep, I'll convert it the morning :smiley:

File: src/webxr/ARButton.ts (hunk 3 above)
Comment (DylanTackoor) [reply src/webxr/ARButton.ts:91]: makes sense to me, will update

File: src/webxr/ARButton.ts (hunk 1 above)
Comment (DylanTackoor) [reply src/webxr/ARButton.ts:3]: will do!

**Comment by CodyJasonBennett** [comment]
Continuing in #241. three.js has broken a lot of stuff around WebXR and its renderers since.

**Pull request closed without merging**